ggu/main.c

```diff
@@ -175,11 +192,38 @@ process_single_packet(struct rte_mbuf *pkt, const struct ggu_config *ggu_conf)
 		if (memcmp(ip6hdr->dst_addr,
 				ggu_conf->net->back.ip6_addr.s6_addr,
 				sizeof(ip6hdr->dst_addr)) != 0) {
+			RTE_LOG(NOTICE, GATEKEEPER,
+				"ggu: received an IPv6 packet destinated to other host!\n");
 			return;
 		}
 
-		udphdr = (struct udp_hdr *)&ip6hdr[1];
+		nexthdr = ip6hdr->proto;
+
+		udp_offset = ipv6_skip_exthdr(ip6hdr, pkt->data_len -
+			sizeof(struct ether_hdr), &nexthdr);
+		if (udp_offset < 0) {
+			RTE_LOG(ERR, GATEKEEPER,
+				"ggu: received non-UDP packets, IPv6 ntuple filter bug!\n");
```

Comment:
The error message is not accurate; it's identical to the following message, which is accurate.